lightningd/channel.c

```diff
@@ -174,7 +174,9 @@ struct channel *new_channel(struct peer *peer, u64 dbid,
 			    const struct pubkey *future_per_commitment_point,
 			    u32 feerate_base,
 			    u32 feerate_ppm,
-			    const u8 *remote_upfront_shutdown_script)
+			    const u8 *remote_upfront_shutdown_script,
+				secp256k1_ecdsa_signature *remote_ann_node_sig,
+				secp256k1_ecdsa_signature *remote_ann_bitcoin_sig)
```

Comment:
@rustyrussell should these be marked as `TAKES` if we `tal_steal` them below? I can never remember the semantics of `TAKES` :wink:

Reply:
I find we marked as `TAKES` if we `tal_dup`, and there's some similar cases in this function:
we didn't marked `scid`, `last_tx`, `last_sent_commit` as `TAKES` and we tal_steal them below,
but we marked `transient_billboard` as `TAKES` because we `tal_strdup` it below.

Reply:
Rusty replied me that `tal_steal` doesn't need to be marked as `TAKES`, and `tal_dup` can be marked  :)